src/firewall/functions.py

```diff
@@ -311,6 +311,17 @@ def checkIPnMask(ip):
                 return False
     return True
 
+def stripNonPrintableCharacters():
+    NOPRINT_TRANS_TABLE = {
+        i: None for i in range(0, sys.maxunicode + 1) if not chr(i).isprintable()
+    }
```

Comment:
The table should still be initialized at the file level. Otherwise we regenerate it every time this function is called. I expect this function to call `rule_str.translate()` with `NOPRINT_TRANS_TABLE` as an argument.